spec/controllers/api/v1/users_controller_spec.rb

```diff
@@ -119,51 +120,44 @@
   end
 
   describe "PUT user/1" do
```

Comment:
I think the structure of the `describe "PUT user/`" specs needs to be reworked. I realise this expands the scope of the ticket a bit but I think it's very worthwhile as it's a key area to be confident in.

Here are some thoughts:

```
context "PUT user/1" do
  context "as a user with can_read_and_manage_users permission" do
    it "I can edit first_name of another user"...
    it "I can set the roles of another user"...
  end

  context "as a user when I edit my own details" do
    it "I can change my first name"
    it "I cannot change my roles"
    it "I cannot change the first name of another user"
  end
end
```

I would then move the specs for `create_or_remove_user_roles` from the User controller to the User model spec file

```
context "create_or_remove_user_roles" do
  context "when new user role is added"
  context "when a role is removed (not present in params)"
  context "when params are nil"
end
```

```